### PR TITLE
feat: implement a Cairo struct to represent NIP-01 event

### PR DESCRIPTION
**Pull request by kateberryd**
implement a Cairo struct to represent NIP-01 event

### CODE REVIEW

**Review by maciejka — COMMENTED** [review]
Please make sure field types match types specified in NIP-01:
```
{
  "id": <32-bytes lowercase hex-encoded sha256 of the serialized event data>,
  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
  "created_at": <unix timestamp in seconds>,
  "kind": <integer between 0 and 65535>,
  "tags": [
    [<arbitrary string>...],
    // ...
  ],
  "content": <arbitrary string>,
  "sig": <64-bytes lowercase hex of the signature of the sha256 hash of the serialized event data, which is the same as the "id" field>
}
```

File: onchain/src/nip01.cairo [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#[derive(Copy, Drop)]
+struct NIP01Event {
+    id: felt252,
+    pubkey: felt252,
```
Comment (maciejka) [inline onchain/src/nip01.cairo:4]: use different type, `pubkey` is 32 bytes, `felt` is 31

File: onchain/src/nip01.cairo [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    created_at: u64,
+    kind: u16,
+    tags: List[List[felt252]]
+    content: ByteArray,
+    sig: felt252
```
Comment (maciejka) [inline onchain/src/nip01.cairo:9]: pubkey is 64 bytes, felt is 31

File: onchain/src/nip01.cairo [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#[derive(Copy, Drop)]
+struct NIP01Event {
+    id: felt252,
+    pubkey: felt252,
+    created_at: u64,
+    kind: u16,
+    tags: List[List[felt252]]
```
Comment (maciejka) [inline onchain/src/nip01.cairo:7]: What kind of `List` is this?

**Review by maciejka — CHANGES_REQUESTED** [review]

File: onchain/src/nip01.cairo [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#[derive(Copy, Drop)]
+struct NIP01Event {
+    id: u256,
+    pubkey: u256,
+    created_at: u64,
+    kind: u16,
+    tags: Array<ByteArray>,
```
Comment (maciejka) [inline onchain/src/nip01.cairo:7]: `tags` should be an array of arrays of `ByteArray`s.

**Review by maciejka — CHANGES_REQUESTED** [review]

File: onchain/src/nip01.cairo (hunk 4 above)
Comment (kateberryd) [reply onchain/src/nip01.cairo:7]: Alright

**Review by maciejka — APPROVED** [review]

**Pull request merged**